File: frontend/src/types/agent.ts
/**
 * agent.ts — TypeScript types pentru Agent BIM SSE events.
 */

/** SSE event: agentul apelează un tool */
export interface ToolCallEvent {
  type: "tool_call";
  tool_name: string;
  tool_input: Record<string, unknown>;
  call_id: string;
}

/** SSE event: rezultatul execuției unui tool */
export interface ToolResultEvent {
  type: "tool_result";
  call_id: string;
  tool_name: string;
  result: Record<string, unknown>;
  duration_ms: number;
}

/** SSE event: text de la agent */
export interface TextDeltaEvent {
  type: "text_delta";
  content: string;
}

/** SSE event: eroare */
export interface ErrorEvent {
  type: "error";
  message: string;
}

/** SSE event: agentul a terminat */
export interface DoneEvent {
  type: "done";
}

/** SSE event: metadata conversație (id, titlu) */
export interface ConversationMetaEvent {
  type: "conversation_meta";
  conversation_id: number;
  title: string;
  is_new: boolean;
}

export type AgentSSEEvent =
  | ToolCallEvent
  | ToolResultEvent
  | TextDeltaEvent
  | ErrorEvent
  | DoneEvent
  | ConversationMetaEvent;

/** Sumar conversație (pentru sidebar) */
export interface ConversationSummary {
  id: number;
  project_id: number;
  title: string;
  message_count: number;
  created_at: string;
  updated_at: string;
}

/** Mesaj din DB */
export interface SavedMessage {
  id: number;
  sequence_num: number;
  role: string;
  content: string;
  tool_steps: ToolStep[] | null;
  created_at: string;
}

/** Conversație detaliată (cu mesaje) */
export interface ConversationDetail {
  id: number;
  project_id: number;
  title: string;
  created_at: string;
  updated_at: string;
  messages: SavedMessage[];
}

/** Un pas de tool call cu rezultat asociat */
export interface ToolStep {
  call_id: string;
  tool_name: string;
  tool_input: Record<string, unknown>;
  result?: Record<string, unknown>;
  duration_ms?: number;
  status: "running" | "completed" | "error";
}

/** Un mesaj din conversația cu agentul */
export interface AgentMessage {
  id: number;
  role: "user" | "assistant" | "system";
  content: string;
  toolSteps?: ToolStep[];
}

/** Tool name → human-readable label */
export const TOOL_LABELS: Record<string, string> = {
  get_project_info: "Informații proiect",
  get_project_context: "Fișa BEP",
  generate_bep: "Generare BEP",
  verify_bep: "Verificare BEP",
  export_bep_docx: "Export DOCX",
  update_project_context: "Actualizare fișă",
  get_verification_history: "Istoric verificări",
  search_bim_standards: "Căutare standarde",
  analyze_ifc_model: "Analiză model IFC",
  list_document_versions: "Versiuni documente",
  compare_bep_versions: "Comparare versiuni BEP",
  get_audit_trail: "Jurnal activități",
  get_project_health_check: "Verificare sănătate proiect",
};

File: backend/app/schemas/agent.py
"""
agent.py — Pydantic schemas pentru Agent BIM chat endpoint și conversații.
"""

from __future__ import annotations

from pydantic import BaseModel, Field


class AgentChatRequest(BaseModel):
    """Request body pentru agent chat SSE endpoint."""
    message: str = Field(
        ...,
        min_length=1,
        description="Mesajul utilizatorului către agent",
    )
    conversation_id: int | None = Field(
        default=None,
        description="ID-ul conversației existente (None = creează nouă)",
    )
    conversation_history: list[dict] = Field(
        default_factory=list,
        description=(
            "Istoricul conversației. Fiecare element: "
            '{"role": "user"|"assistant", "content": "..."}'
        ),
    )


# ── Conversation schemas ─────────────────────────────────────────────────────

class ConversationCreate(BaseModel):
    """Request body pentru crearea unei conversații noi."""
    title: str = Field(default="Conversație nouă", max_length=255)


class ConversationRead(BaseModel):
    """Sumarul unei conversații (pentru lista din sidebar)."""
    id: int
    project_id: int
    title: str
    message_count: int
    created_at: str
    updated_at: str


class MessageRead(BaseModel):
    """Un mesaj din conversație."""
    id: int
    sequence_num: int
    role: str
    content: str
    tool_steps: list[dict] | None = None
    created_at: str


class ConversationDetailRead(BaseModel):
    """Conversație completă cu mesaje."""
    id: int
    project_id: int
    title: str
    created_at: str
    updated_at: str
    messages: list[MessageRead]
